src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -593,37 +638,73 @@ private static TextSpan GetSpan(Stack<SyntaxNodeOrToken> stack, int first, int l
             return TextSpan.FromBounds(start, end);
         }
 
-        private static Stack<SyntaxNodeOrToken> Combine(Stack<SyntaxNodeOrToken> first, Stack<SyntaxNodeOrToken> next)
+        private static TextSpan GetSpan(Queue<SyntaxNodeOrToken> queue, int first, int length)
+        {
+            int start = -1, end = -1, i = 0;
+            foreach (var n in queue)
+            {
+                if (i == first)
```

Comment:
man, it seems super unfortunate that there's no queue-like structure that also allows indexed access.